db_setup_docker.sql

```diff
@@ -1,3 +1,4 @@
 CREATE USER postgres with password 'postgres';
 GRANT ALL PRIVILEGES ON DATABASE serviceapitests TO postgres;
 GRANT ALL PRIVILEGES ON DATABASE serviceapitests TO kas_fleet_manager;
+ALTER SYSTEM SET max_connections=200;
```

Comment:
```suggestion
ALTER SYSTEM SET max_connections=200;

```
adding in the new line

Reply:
Could you explai why has this been added?

Reply:
> Could you explain why has this been added?

Adding a newline at the end of the file or the addition of the SQL statement?

Reply:
the addition of the sql statement

Reply:
Ah right. This comment https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1279#discussion_r961580190 from @JameelB  explains it.

Reply:
How was the test made? I tried it locally without this change and this is what I observe:
```
msoriano@localhost:~/go/src/github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager (main)$ git diff
diff --git a/db_setup_docker.sql b/db_setup_docker.sql
index c3fd21ff..f1d4c905 100644
--- a/db_setup_docker.sql
+++ b/db_setup_docker.sql
@@ -1,4 +1,3 @@
 CREATE USER postgres with password 'postgres';
 GRANT ALL PRIVILEGES ON DATABASE serviceapitests TO postgres;
 GRANT ALL PRIVILEGES ON DATABASE serviceapitests TO kas_fleet_manager;
-ALTER SYSTEM SET max_connections=200;

msoriano@localhost:~/go/src/github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager (main)$ make db/setup
./scripts/local_db_setup.sh
bb89ade25fd27f66652dd58daf73f2b47707f1ad37327883250e07682ff30bfc
54a43ca2e401424c052fa9f92b7559e9a69844d60983d9238474033f75123571

msoriano@localhost:~/go/src/github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager (main)$ make db/migrate
go run ./cmd/kas-fleet-manager migrate
I0902 16:13:50.213081  341164 environment.go:58] Environment variable "OCM_ENV" not specified, using default "development"
I0902 16:13:50.214918  341164 cmd.go:19] Migration starting
I0902 16:13:50.947170  341164 cmd.go:22] Database has 62 migrations applied

msoriano@localhost:~/go/src/github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager (main)$ docker exec -it  kas-fleet-manager-db sh
# psql serviceapitests kas_fleet_manager
psql (13.7 (Debian 13.7-1.pgdg110+1))
Type "help" for help.

serviceapitests=# show max_connections;
 max_connections 
-----------------
 200
(1 row)
```

cc @JameelB

Reply:
For the local development/test database - the maximum connection is controlled by https://github.com/JameelB/kas-fleet-manager/blob/f42a96e031bc0d42f06890b0f4f9452ea49789aa/scripts/local_db_setup.sh#L22

The SQL statement is what controls the database used in the midstream PR checks, and nightly run. The file is used as an an init-db file in the dockerfiles used to setup the environment to run the integration tests: 
1. https://github.com/JameelB/kas-fleet-manager/blob/f42a96e031bc0d42f06890b0f4f9452ea49789aa/docker/Dockerfile_template_mocked#L44
2. https://github.com/JameelB/kas-fleet-manager/blob/f42a96e031bc0d42f06890b0f4f9452ea49789aa/docker/Dockerfile_template#L53

Reply:
This is for the [pr_check.sh](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pr_check.sh) script used by the midstream prs. The make target uses a different script, [local_db_setup.sh](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/scripts/local_db_setup.sh)

You'll need to run `OCM_ENV=integration ./pr_check.sh` with mas sso stage credentials specified in the env vars. It should bring up a docker container. The screenshot I took from the comment above was from the docker container.

Reply:
To test the midstream CI dockerfiles:
change 
- https://github.com/JameelB/kas-fleet-manager/blob/f42a96e031bc0d42f06890b0f4f9452ea49789aa/docker/Dockerfile_template#L1 and use `postgres` instead of `quay.io/app-sre/postgres`
- and make sure you've the mas-sso environment variables set 
- then run the `pr_check.sh` file

Once the container is running, you can run `psql` against it to observe the max_connections set using this added SQL statement

Reply:
Thanks I will try it :+1:

Reply:
An additional/side comment:

In the dockerfile `quay.io/app-sre/postgres` is used. I understand that means that it uses the `latest` tag. I also understand that that image is a mirror of https://hub.docker.com/_/postgres/. The `latest` tag is currently using PostgreSQL 14.
You can verify this by running:
```sh
msoriano@localhost:~/go/src/github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager (main)$ docker pull docker.io/postgres
Using default tag: latest
latest: Pulling from library/postgres
...

msoriano@localhost:~/go/src/github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager (main)$ docker run -it --rm postgres --version
postgres (PostgreSQL) 14.5 (Debian 14.5-1.pgdg110+1)
```

We should update the Dockerfiles to use postgresql 13 there too to be consistent. I am assuming 13 has been chosen due to the db we are using for stage and prod is postgresql 13 (let me know if that's not accurate)

Reply:
> In the dockerfile quay.io/app-sre/postgres is used. I understand that means that it uses the latest tag. I also understand that that image is a mirror of https://hub.docker.com/_/postgres/. The latest tag is currently using PostgreSQL 14.

Yes. This is correct. We should probably use a specific tag and not a floating one. Given that the quay repo mirrors the postgres upstream one, we should be good to use whichever tag that's in upstream.

> We should update the Dockerfiles to use postgresql 13 there too to be consistent. 

+1

> I am assuming 13 has been chosen due to the db we are using for stage and prod is postgresql 13 (let me know if that's not accurate)

If we want to be exact, 12.7 is the one we are using in our envs.

Reply:
Regarding my previous comment I've been looking at stage/prod and if I am understanding correctly the configurations it seems we are using PostgreSQL 12.x. In that case my previous comment applies to PostgreSQL 12 but we should also change it in the other places of the repository.

cc @machi1990 do you know what version is used for stage/prod?

Reply:
I found why `-N 200` didn't "work" @JameelB @machi1990:

Part of the Dockerfile build process copies the file `pr_check_docker.sh` from the project root into the `/docker-entrypoint-initdb.d/` directory of the container. Contents placed in that directory are automatically executed when the image is run.
If you pay attention to that file some of the steps are stopping the postgresql server and starting it again with some hardcoded flags, which basically ignores all provided flags.

Setting the max connections in the db_setup_docker.sql script is also a fortunate coincidence. Setting it does not change the connections until postgresql server is restarted. Due to the `pr_check_docker.sh` script stops and starts the server the change is made effective. Setting it through that way also does not make it 100% permanent. What it does is it stores the setting in `postgresql.auto.conf` which is different than the permanent `postgresql.conf` although it's also read aside from that. If someone were to perform an alter system reset or set the parameter to default that would be lost. For more detail on behavior of alter system for postgresql see: https://www.postgresql.org/docs/current/sql-altersystem.html.

Reply:
yes, the max_connection is a config that needs to be set at start/otherwise postgres needs to restart which luckily the script was already doing :) 

I think it should be fine to have it in the auto.conf, it's read last i believe and takes precendence over postgresql.conf. There also shouldn't be anything that resets the configuration of the db and restart it again during test runs :thinking:

Reply:
:+1: it's not clean but it does the work so let's keep it that way.
What we need to change though is the postgresql version used in several places